src/main/java/com/duke/bot/TaskList.java

```diff
@@ -0,0 +1,83 @@
+package com.duke.bot;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Represents the data structure that stores the tasks for Duke Bot.
+ */
+public class TaskList {
+    List<Task> tasks;
+
+    private TaskList() {
+        this(0);
+    }
+
+    private TaskList(int n) {
+        tasks = new ArrayList<>(n);
+    }
```

Comment:
I think this is a good approach as for the current Duke project, we most of the time only have less than 10 entries in the taskList. This could help to save memory allocation. But at the same time, this could be a trade-off to time-consuming as it would be required ArrayList to keep resize when adding the new element to it.